EuclideanGeometry/Foundation/Construction/Triangle/Orthocenter.lean

```diff
@@ -10,7 +11,15 @@ variable {P : Type _} [EuclideanPlane P]
 
 def TriangleND.Height1 (tr_nd : TriangleND P) : SegND P := sorry
 
-structure IsOrthocenter (tr_nd : TriangleND P) (H : P) : Prop where
+-- this takes care of corner cases such as right triangles, where AH ⟂ BC runs into trouble since A = H
+def IsOrthocenter (tr_nd : TriangleND P) (H : P) : Prop := inner (VEC tr_nd.point₁ H) (VEC tr_nd.point₂ tr_nd.point₃) = (0 : ℝ) ∧ inner (VEC tr_nd.point₂ H) (VEC tr_nd.point₃ tr_nd.point₁) = (0 : ℝ) ∧ inner (VEC tr_nd.point₃ H) (VEC tr_nd.point₁ tr_nd.point₂) = (0 : ℝ)
```

Comment:
1. Please write this as a structure
```
structure IsOrthocenter ... : Prop where

```
to avoid usage of plural `and`. 

2. Is it possible to use `perpendicular`, or `perp foot` or `LiesOn Height1` to define orthocenter? I do hope that there will be a "uniform" way of defining each center of a triangle. This needs a further discussion.

Reply:
3. Please write a plan file (possibly using overleaf) if you are going to build a whole new file, thank you! You may find some example of plan file in the folder `PlanFiles`.

Reply:
LiesOn Height1 will work, but the necessary lemmas are not ready yet.